simapp/sim_test.go

```diff
@@ -46,148 +51,95 @@ func interBlockCacheOpt() func(*baseapp.BaseApp) {
 }
 
 func TestFullAppSimulation(t *testing.T) {
-	if !FlagEnabledValue {
+	config, db, dir, logger, skip, err := SetupSimulation("leveldb-app-sim", "Simulation")
+	if skip {
 		t.Skip("skipping application simulation")
 	}
-
-	var logger log.Logger
-	config := NewConfigFromFlags()
-	config.ChainID = helpers.SimAppChainID
-
-	if FlagVerboseValue {
-		logger = log.TestingLogger()
-	} else {
-		logger = log.NewNopLogger()
-	}
-
-	var db dbm.DB
-	dir, _ := ioutil.TempDir("", "goleveldb-app-sim")
-	db, _ = sdk.NewLevelDB("Simulation", dir)
+	require.NoError(t, err, "simulation setup failed")
 
 	defer func() {
 		db.Close()
-		os.RemoveAll(dir)
+		require.NoError(t, os.RemoveAll(dir))
 	}()
 
 	app := NewSimApp(logger, db, nil, true, FlagPeriodValue, fauxMerkleModeOpt)
 	require.Equal(t, "SimApp", app.Name())
 
-	// Run randomized simulation
+	// run randomized simulation
 	_, simParams, simErr := simulation.SimulateFromSeed(
-		t, os.Stdout, app.BaseApp, AppStateFn(app.Codec(), app.sm),
+		t, os.Stdout, app.BaseApp, AppStateFn(app.Codec(), app.SimulationManager()),
 		SimulationOperations(app, app.Codec(), config),
 		app.ModuleAccountAddrs(), config,
 	)
 
-	// export state and params before the simulation error is checked
-	if config.ExportStatePath != "" {
-		err := ExportStateToJSON(app, config.ExportStatePath)
-		require.NoError(t, err)
-	}
-
-	if config.ExportParamsPath != "" {
-		err := ExportParamsToJSON(simParams, config.ExportParamsPath)
-		require.NoError(t, err)
-	}
-
+	// export state and simParams before the simulation error is checked
+	err = CheckExportSimulation(app, config, simParams)
+	require.NoError(t, err)
 	require.NoError(t, simErr)
```

Comment:
Can we have this call before `CheckExportSimulation`?

Reply:
> Can we have this call before CheckExportSimulation?

`// export state and simParams before the simulation error is checked`

The reason bc is in that order is to be able to export the params and state before the error is checked. Otherwise, you will see the error but won't have those exported files to be able to reproduce the error or evaluate it.